<?xml version="1.0" encoding="utf-8"?>
<launch>

    <param name="/use_sim_time" type="bool" value="false"/>

    <arg name="topic_trigger_insert" value="/rnw/trigger/insert" />
    <arg name="topic_trigger_zigzag" value="/rnw/trigger/zigzag" />
    <arg name="topic_trigger_topple" value="/rnw/trigger/topple" />
    <arg name="topic_trigger_rnw" value="/rnw/trigger/rnw" />
    <arg name="topic_trigger_adjust_nutation" value="/rnw/trigger/adjust_nutation" />
    <arg name="topic_trigger_adjust_grip" value="/rnw/trigger/adjust_grip" />

    <arg name="topic_dbg_trigger" value="/rnw/trigger/debug" />

    <rosparam command="load" file="$(find rnw_ros)/config/rnw_config.yaml"/>

    <node pkg="rnw_ros" type="pub_mock_uav_node" name="pub_mock_uav_node" output="screen" required="true"/>

    <node pkg="mocap_optitrack" type="mocap_node" name="mocap_node" respawn="false" launch-prefix="" required="true" output="log">
        <rosparam file="$(find mocap_optitrack)/config/mocap.yaml" command="load" />
    </node>

    <node pkg="pos_vel_mocap" type="pos_vel_mocap" name="pos_vel_mocap" output="screen" required="true">
        <param name="publish_uav_odom" type="bool" value="false"/>
        <rosparam file="$(find pos_vel_mocap)/config/calib.yaml" command="load" />
    </node>

    <node pkg="rnw_ros" type="pub_cone_state_node" name="pub_cone_state_node" required="true" output="log">
        <param name="cut_euler_velocity" type="bool" value="true"/>
        <param name="max_euler_velocity" value="10"/>
    </node>

    <node pkg="joy" type="joy_node" name="joy_node" output="screen">
        <param name="~autorepeat_rate" value="0" />
    </node>

    <node pkg="rnw_ros" type="gamepad_trigger_node" name="gamepad_trigger_node" output="screen">
        <remap from="~B" to="$(arg topic_dbg_trigger)"/>
        <remap from="~A" to="$(arg topic_trigger_rnw)"/>
        <remap from="~X" to="$(arg topic_trigger_zigzag)"/>
        <remap from="~Y" to="$(arg topic_trigger_topple)"/>
        <remap from="~LT" to="$(arg topic_trigger_adjust_nutation)"/>
        <remap from="~RT" to="$(arg topic_trigger_adjust_grip)"/>
    </node>

    <node pkg="rnw_ros" type="visualize_cone_node" name="visualize_cone_node" output="screen" required="true">
        <param name="cone_color_r" type="double" value="0"/>
        <param name="cone_color_g" type="double" value="1"/>
        <param name="cone_color_b" type="double" value="0"/>
    </node>

    <node pkg="rnw_ros" type="rnw_controller_node" name="rnw_controller_node" output="screen"/>

    <node pkg="rnw_ros" type="visualize_traj_node" name="traj_visualization" output="screen" required="true">
        <param name="lift_dt" type="double" value="0.05"/>
        <param name="length_g" type="double" value="0.2"/>
        <param name="clear_after_n_sec" type="double" value="0"/>
    </node>

    <node name="rviz" pkg="rviz" type="rviz" output="log" args="-d $(find rnw_ros)/config/traj.rviz"/>

<!--    <node pkg="rqt_multiplot" type="rqt_multiplot" name="rqt_multiplot" output="log" required="true"-->
<!--          args="&#45;&#45;multiplot-config $(find rnw_ros)/config/rqt_multiplot_grip.xml &#45;&#45;multiplot-run-all"/>-->


</launch>
